restapi/src/main/java/io/stargate/web/docsapi/service/DocsShredder.java

```diff
@@ -177,42 +173,40 @@ private Object convertToBackendBooleanValue(boolean value, boolean numericBoolea
       String key,
       boolean patching,
       boolean numericBooleans) {
-    JsonPath jsonPath =
-        JsonPathCompiler.compile(
-            DocsApiUtils.convertJsonToBracketedPath(parsingContext.getJsonPath()));
+    String bracketedPath = DocsApiUtils.convertJsonToBracketedPath(parsingContext.getJsonPath());
     Map<String, Object> bindMap = DocsApiUtils.newBindMap(path, config.getMaxDepth());
 
     bindMap.put("key", key);
-
-    Iterator<PathOperator> it = jsonPath.iterator();
     String leaf = null;
     int i = path.size();
 
     String[] unboundPaths = new String[config.getMaxDepth()];
     List<String> firstLevelKeys = new ArrayList<>();
-    while (it.hasNext()) {
+    String[] bracketedPathArray = DocsApiUtils.bracketedPathAsArray(bracketedPath);
+    for (int j = 0; j < bracketedPathArray.length; j++) {
       if (i >= config.getMaxDepth()) {
         throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_GENERAL_DEPTH_EXCEEDED);
       }
 
-      PathOperator op = it.next();
-      String pathValue = op.toString();
+      String pathValue = bracketedPathArray[j];
+      boolean isArrayValue = !pathValue.startsWith("'");
 
-      if (!pathValue.equals("$")) {
-        // pathValue always starts and ends with a square brace because of
-        // DocsApiUtils#convertJsonToBracketedPath
-        String innerPath =
-            DocsApiUtils.convertEscapedCharacters(pathValue.substring(1, pathValue.length() - 1));
-        if (isPatchingWithArrayValue(i, path.size(), op, patching)) {
-          throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_PATCH_ARRAY_NOT_ACCEPTED);
-        }
-        if (isAtTopLevel(i, path.size())) {
-          firstLevelKeys.add(innerPath);
-        }
-        String convertedPath = convertPathValueForArrays(innerPath, op);
-        unboundPaths[i++] = convertedPath;
-        leaf = convertedPath;
+      // pathValue always starts and ends with a square brace because of
+      // DocsApiUtils#convertJsonToBracketedPath
+      String innerPath = DocsApiUtils.convertEscapedCharacters(pathValue);
```

Comment:
can we not do this for arrays, I guess there is no need? Should be optimized by a compiler though.. Ignore..